### ISSUE TITLE
subscribe to empty array

### ISSUE THREAD

**Issue by mepdr** [mqttjs/MQTT.js#496]
Subscribing to an empty array causes an application disconnection and the subscribe callback is not called.
The testing app below prints: 
```
offline 
close
reconnect
```

```js
var mqtt = require('mqtt')
var c  = mqtt.connect('ws://test.mosquitto.org:8080/mqtt')

c.on("message", function(topic, payload) {
    console.log(topic);
});

c.on("reconnect", function() {
    console.log("reconnect");
});

c.on("close", function() {
    console.log("close");
});

c.on("offline", function() {
    console.log("offline");
});

c.on("error", function() {
    console.log("error");
});

c.subscribe([], {qos:1}, function (err, granted) {
    console.log(err, granted);
});
```

**Comment by mcollina** [comment]
Thanks for reporting, would you like to send a PR to fix this issue?

**Comment by johnzeng** [comment]
According to MQTT 3.1.1 protocol ,  The payload of a SUBSCRIBE packet MUST contain at least one Topic Filter/QoS pair. A SUBSCRIBE packet with no payload is a protocol violation.

**Comment by mcollina** [comment]
I think we should error on the client side, before  sending a message to the server.

**Comment by yohei1126** [comment]
Has this issue already closed?

**Comment by mcollina** [comment]
Fixed by #497

**Comment by mcollina** [comment]
Thanks!!!
